modules/compute-mig/main.tf

```diff
@@ -152,22 +152,24 @@ locals {
 }
 
 resource "google_compute_per_instance_config" "default" {
-  count = var.stateful_disk_instance == null ? 0 : 1
-  zone  = var.location
+  for_each = try(var.stateful_config.per_instance_config, tomap({}))
+  #for_each = var.stateful_config && var.stateful_config.per_instance_config == null ? {} : length(var.stateful_config.per_instance_config)
+  zone = var.location
   # terraform error, solved with locals
   #instance_group_manager           = var.regional ? google_compute_region_instance_group_manager.default : google_compute_instance_group_manager.default
   instance_group_manager           = local.instance_group_manager[0].id
-  name                             = var.name_instance_config
+  name                             = each.key
   project                          = var.project_id
-  minimal_action                   = var.minimal_action
-  most_disruptive_allowed_action   = var.most_disruptive_allowed_action
-  remove_instance_state_on_destroy = var.remove_instance_state_on_destroy
+  minimal_action                   = each.value.update_config.minimal_action
```

Comment:
Would it make sense to guard against the user passing a `null` value for `update_config`? I skimmed the resource docs, and everything seems to be optional except for `name` and `instance_group_manager`. If that's so, we should wrap in `try`.